modules/log/logger.ts

```diff
@@ -0,0 +1,124 @@
+import _ from 'lodash';
+import Winston from 'winston';
+
+
+export default class Logger {
+
+	public log: any;
+
+	private defaultWidth: number = 0;
+
+	constructor() {
+		this.log = new Winston.Logger({
+			transports: [
+				new Winston.transports.Console({
+					level: 'silly',
+					handleExceptions: true,
+					json: false,
+					colorize: false,
+				}),
+			],
+		});
+	}
+
+	public setLogLevel(level: string): void {
+		level = checkLevel(level);
+		this.log.transports.console.level = level;
+		this.log.info(`Log level set to ${level}`);
+	}
+
+	public setUseColor(useColor: boolean) {
+		this.log.transports.console.colorize = useColor === true;
+		this.log.info(`Logger will use color: ${useColor}`);
+	};
+
+
+	public silly(msg: string) {
+		this.log.silly(msg);
+	}
+
+	public debug(msg: string) {
+		this.log.debug(msg);
+	}
+
+	public verbose(msg: string) {
+		this.log.verbose(msg);
+	}
+
+	public info(msg: string) {
+		this.log.info(msg);
+	}
+
+	public error(msg: string) {
+		this.log.error(msg);
+	}
+
+	public profile(msg: string) {
+		this.log.profile(msg);
+	}
+
+	public warn(msg: string) {
+		this.log.warn(msg);
+	}
+
+	public writeBoxedLine(string: string, width: number, padchar: string, level: string) {
+		level = checkLevel(level);
+		if (!width) width = this.defaultWidth;
+		if (!padchar) padchar = ' ';
+		let cWidth = width - 4;
+		let words = string.split(' ');
+		let rows = [];
+		let c = 0;
+		_.forEach(words, function (word, index) {
+			if (!rows[c]) rows[c] = '';
+			if (rows[c].length + word.length + 1 > cWidth) {
+				c++;
+				rows[c] = '';
+			}
+			rows[c] += word + ' ';
+		});
+
+		// Lodash messes with this/self. self is set to the Builder object for some reason.
+		let logObject = this.log;
+		_.forEach(rows, function (row, index) {
+			logObject.log(level, '║ ' + pad(row.trimRight(), cWidth, padchar) + ' ║');
+		});
+	};
+
+
+	public writeBoxTop(width, level) {
```

Comment:
There are several functions here that are missing parameter types. You should be getting linter warnings about not using an "implicit any".